### PR TITLE
Matrix, Quaternion and lowpass filter cleanup

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
Merge to be discussed - will submit downstreams in a minute.

File: tests/matrix.cpp [hunk 1]
```diff
@@ -76,3 +75,10 @@ BOOST_AUTO_TEST_CASE( lookat )
     BOOST_CHECK_EQUAL( lookAt, newLookAt );
     BOOST_CHECK_EQUAL( up, newUp );
 }
+
+// Verify code by instantiating some templates:
+template class vmml::Matrix< 1, 1, float >;
+template class vmml::Matrix< 2, 2, double >;
+template class vmml::Matrix< 3, 3, short >;
+template class vmml::Matrix< 3, 4, int >;
+//template class vmml::Matrix< 4, 4, unsigned >;
```
Comment (tribal-tec) [inline tests/matrix.cpp]: commented?

**Comment by tribal-tec** [comment]
vmmlib/matrix.hpp
755 additions, 2,362 deletions not shown because the diff is too large. Please use a local Git client to view these changes.

Offline review next week or you merge already and review later. Your call.

File: tests/matrix.cpp (hunk 1 above)
Comment (eile) [inline tests/matrix.cpp]:
```
../vmmlib/vmmlib/matrix.hpp:620:13: error: call to 'abs' is ambiguous
        if( std::abs( array[ i ] - other.array[ i ]) > tolerance )
            ^~~~~~~~
../vmmlib/tests/matrix.cpp:84:22: note: in instantiation of member function 'vmml::Matrix<4, 4, unsigned int>::equals' requested here
template class vmml::Matrix< 4, 4, unsigned >;
                     ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/include/stdlib.h:129:5: note: candidate function
int abs(int) __attribute__((const));
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cstdlib:159:75: note: candidate function
inline __attribute__ ((__visibility__("hidden"), __always_inline__)) long abs( long __x) noexcept {return labs(__x);}
                                                                          ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cstdlib:161:80: note: candidate function
inline __attribute__ ((__visibility__("hidden"), __always_inline__)) long long abs(long long __x) noexcept {return llabs(__x);}
                                                                               ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cmath:660:1: note: candidate function
abs(float __x) noexcept {return fabsf(__x);}
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cmath:664:1: note: candidate function
abs(double __x) noexcept {return fabs(__x);}
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cmath:668:1: note: candidate function
abs(long double __x) noexcept {return fabsl(__x);}
^
```

**Comment by eile** [comment]
After release - review next week is sufficient.

File: tests/matrix.cpp (hunk 1 above)
Comment (bilgili) [inline tests/matrix.cpp]: Some maths operations with unsigned are not commutative on type T. Therefore other than keeping data most maths operations will be ill formed with operations on commutative properties. In short a - b != b - a

**Pull request merged**